<launch>
  <include file="$(find gazebo_ros)launch/empty_world.launch">
      <arg name="world_name" value="$(find example_package)/world/playground.world" />
      <arg name="gui" value="true" />
      <arg name="use_sim_time" value="true" />
      <arg name="paused" value="false" />
  </include>

  <param name="robot_description" command="$(find xacro)/xacro --inorder $(find turtlebot3_description)/urdf/turtlebot3_burger_for_autorace.urdf.xacro" /> 
  <node pkg="gazebo_ros" type="spawn_model" name="spawn_urdf" args="-urdf -model turtlebot3_burger -x 0 -y 0 -z 0 -Y 0 -param robot_description" />
  <include file="$(find turtlebot3_bringup)/launch/turtlebot3_remote.launch"/>

  <!-- Map Server Node -->
  <node pkg="map_server" type="map_server" name="map_server" args="$(find example_package)/maps/my_map.yaml" />
  <node pkg="tf" type="static_transform_publisher" name="static_map_to_odom" args="0 0 0 0 0 0 map odom 100" />

  <node type="rviz" name="rviz" pkg="rviz" args="-d $(find example_package)/config/config.rviz" />

  <node pkg="turtlebot3_teleop" type="turtlebot3_teleop_key" name="turtlebot3_teleop_key" launch-prefix="xterm -e"/>
  
  <!-- Kalman Filter Node-->
  <node pkg="example_package" type="filter_node_Kalman" name="filter_node_Kalman" output="screen"/>

  <!-- EKF Filter Node-->
  <node pkg="example_package" type="filter_node_EKF" name="filter_node_EKF" output="screen"/>

  <!-- Particle Filter Node-->
  <node pkg="example_package" type="filter_node_particle" name="filter_node_particle" output="screen">
      <param name="num_particles" value="300"/>
      <param name="alpha1" value="0.2"/>
      <param name="alpha2" value="0.2"/>
      <param name="alpha3" value="0.1"/>
      <param name="alpha4" value="0.1"/>
  </node>

    <!-- starting rqt_reconfigure Node-->
  <node pkg="rqt_reconfigure" type="rqt_reconfigure" name="rqt_reconfigure" output="screen" />

</launch>
